desktop/app/app-state.js

```diff
@@ -0,0 +1,195 @@
+//
+// This is modified from https://github.com/mawie81/electron-window-state
+//
+
+import electron from 'electron'
+import fs from 'fs'
+import path from 'path'
+import jsonfile from 'jsonfile'
+import deepEqual from 'deep-equal'
+
+export default class AppState {
+  constructor (opts) {
+    this.opts = opts || {}
+    this.state = {}
+    this.eventHandlingDelay = 100
+
+    this.config = Object.assign({
```

Comment:
Is there a reason you're using Object.assign instead of `{...this.opts, file: 'app-state.json'}` ?